src/modROS.lua

```diff
@@ -136,112 +130,55 @@ function ModROS:update(dt)
 end
 
 
--- -- A.1 getPublisher - instantiate publishers for each spec instance, called from RosVehicle.lua
--- function ModROS:getPublisher(topic_name, topic_type)
---     local pub = Publisher.new(self._conx, topic_name, topic_type)
---     return pub
--- end
-
-
--- B.1 sim_time publisher: publish the farmsim time
+-- A.1 sim_time publisher: publish the farmsim time
 function ModROS:publish_sim_time_func()
     local msg = rosgraph_msgs_Clock.new()
     msg.clock = ros.Time.now()
     self._pub_clock:publish(msg)
 end
 
 
--- B.2. odom publisher
+-- A.2. odom publisher
 -- a function to publish ground-truth Pose and Twist of all vehicles (including non-drivable) based on their in-game position and orientation
-function ModROS:publish_veh_func()
+function ModROS:publish_veh_odom_func()
     -- FS time is "frozen" within a single call to update(..), so this
     -- will assign the same stamp to all Odometry messages
     local ros_time = ros.Time.now()
     for _, vehicle in pairs(g_currentMission.vehicles) do
+        -- only if the vehicle is spec_rosVehicle, the pubOdom() can be called
```

Comment:
Can be called, or *is* called?

Should we be filtering for `rosVehicle` specs?

Same question below everywhere the same comment occurs.

Reply:
When I wrote this comment, I hadn't seen yet that we now list prerequisites.

AFAIU, FS won't install the spec if the prerequisites aren't fulfilled, so checks for the correct spec are probably not needed.

Reply:
Why can't I see this comment on the conversation tab?

Reply:
> Can be called, or is called?

is called. The wording was weird.

> Should we be filtering for rosVehicle specs?

In theory, we should, but I've tested and found out that if vehicle is not drivable, the pubOdom() and pubLaserScan() won't be called but there is no error.